src/Reveal.Sdk.Dom.Tests/Data/DataSources/RestDataSourceFixture.cs

```diff
@@ -104,5 +109,70 @@ public void Method_SetsAndGetsValue_WithDifferentInputs(string method)
             Assert.Equal(method, dataSource.Method);
             Assert.Equal(method, dataSource.Properties.GetValue<string>("Method"));
         }
+
+        [Fact]
+        public void RDashDocument_HasCorrectDataSource_WhenLoadFromFile()
+        {
+            // Arrange
+            var filePath = Path.Combine(Environment.CurrentDirectory, "Dashboards", "TestRest.rdash");
+
+            // Act
+            var document = RdashDocument.Load(filePath);
+            var dataSource = document.DataSources.LastOrDefault();
+
+            // Assert
+            Assert.Equal(DataSourceProvider.REST, dataSource.Provider);
+            Assert.NotNull(dataSource.Properties.GetValue<string>("Url"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUseAnonymousAuthentication"));
+            Assert.NotNull(dataSource.Properties.GetValue<string>("_rpUsePreemptiveAuthentication"));
+        }
+
+        [Fact]
+        public void ToJsonString_CreatesFormattedJson_ForRestDataSource()
```

Comment:
Rest will actually add multiple data sources to the DataSources collection depending on the return type (json, excel, or CSV). I believe the DataSpec.DataSourceItem.ResourceItem.DataSourceId would point to it.